docs/README.md

```diff
@@ -399,6 +399,137 @@ const buff = await response.arrayBuffer()
+const token = ...
+const client = await cozy.auth.getClient(new cozy.auth.Client('https://me.cozy.io/', {
+  clientID: '1235'
+}), token)
+```
+
+
+### `cozy.auth.getAuthCodeURL(client, scopes)`
+
+`cozy.auth.getAuthCodeURL` is used to generate the URL on which the user should go to give access to the application with the specified scopes.
+
+It returns an object with the url and a generated random state that should be stored to be matched again for the token exchange phase.
+
+- `client` is a registered `cozy.auth.Client`
+- `scopes` is an array of permission strings formatted as `key:access` (like `files/images:read`)
+
+```js
+const {url, state} = cozy.auth.getAuthCodeURL(client, ['files/images:read'])
+
+// save state and redirect to url
+localStorage.setItem("oauthstate", state)
+document.location.replace(url)
+```
+
+
+### `cozy.auth.getAccessToken(client, state, pageURL)`
+
+`cozy.auth.getAccessToken` is used from the page on which the user should have redirected after authorizing the application.
+
+The method verifies that the specified state and the extracted one match. It then ask the server for a new bearer token and returns it.
+
+- `client` is a registered `cozy.auth.Client`
+- `state` is the previously stored state that is matched against to prevent CSRF attacks
+- `pageURL` is the url of the current page from the which a code and state will be extracted. If empty, `document.location.href` is used
+
+```js
+const client = cozy.auth.getClient(/* ... */)
+const state = localStorage.getItem("oauthstate")
+const pageURL = document.location.href
+const token = cozy.auth.getAccessToken(client, state, pageURL)
+```
+
+
+### `cozy.auth.refreshToken`
+
+`refreshToken` is used to refresh a token that is expired or no more valid.
+
+- `client` is a registered `cozy.auth.Client`
+- `token` is a valid `cozy.auth.Token`
+
+```js
+const newtoken = cozy.auth.refreshToken(client, oldtoken)
+```
+
+
+### `cozy.auth.Client`
+
+`cozy.auth.Client` is a class representing an OAuth client. It can be registered, in which case it is known by the server and has a `clientID` and `clientSecret`.
+
+```
+type Client {
+  clientID: string                // informed by server
+  clientSecret: string            // informed by server
+  registrationAccessToken: string // informed by server
+  url: string;         // mandatory
+  redirectURI: string; // mandatory
+  softwareID: string;  // mandatory
+  softwareVersion: string;
+  clientName: string;  // mandatory
+  clientKind: string;
+  clientURI: string;
+  logoURI: string;
+  policyURI: string;
+}
+```
+
+The constructor type is as follow:
+
+- `url` is a string of the url of the cozy
+- `options` is an object with the same fields as a client object, or is an instance of client
+
+```
+new Client(url, options)
+```
+
+
+### `cozy.auth.Token`
+
+`cozy.auth.Token` is a class representing an OAuth token.
+
+```
+type Token {
+  tokenType: string;
+  accessToken: string;
+  refreshToken: string;
+  scope: string
```

Comment:
I'm a bit puzzled here. What are `tokenType` and `scope`? And there should be a registrationToken (or registrationAccessToken) here, no?